Packs/Cofense-Intelligence/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py

```diff
@@ -0,0 +1,390 @@
+import base64
+
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+import requests
+import traceback
+from typing import Dict, Any
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+VENDOR_NAME = "Cofense Intelligence v2"
+INTEGRATION_NAME = "CofenseIntelligenceV2"
+OUTPUT_PREFIX = 'CofenseIntelligence.Threat'
+RELIABILITY = 'integration_reliability'
+
+SEVERITY_SCORE = {'None': 0, 'Minor': 1, 'Moderate': 2, 'Major': 3}
+VERDICT = {0: 'Unknown', 1: 'Good', 2: 'Suspicious', 3: 'Bad'}
+
+EMAIL_REGEX = r'[^@]+@[^@]+\.[^@]+'
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
+    """
+
+    def threat_search_call(self, ip=None, file=None, email=None, url=None, string=None):
+        params = {}
+        if ip:
+            params['ip'] = ip
+        elif email:
+            params['watchListEmail'] = email
+        elif file:
+            params['allMD5'] = file
+        elif url:
+            params['urlSearch'] = url
+        elif string:
+            params['extractedString'] = string
+        return self._http_request(method='POST', url_suffix='/threat/search', params=params)
+
+
+def create_threat_md_row(threat: Dict, severity_level: int = None):
+    threat_row = {"Threat ID": threat.get("id", ""),
+                  "Threat Types": "\n".join(
+                      [m.get("description", "") for m in threat.get("malwareFamilySet", [])]),
+                  "Executive Summary": threat.get("executiveSummary", ""),
+                  "Campaign": threat.get("label", ""),
+                  "Last Published": epochToTimestamp(threat.get("lastPublished"))}
+    if severity_level:
+        threat_row["Verdict"] = VERDICT.get(severity_level)
+    return threat_row
+
+
+def threats_analysis(threats: List, indicator: str, threshold: str):
+    threshold_score = SEVERITY_SCORE.get(threshold)
+    if not threshold_score:
+        raise Exception(
+            f'Cofense error: Invalid threshold value: {threshold}. Valid values are: None, Minor, Moderate or Major')
+
+    md_data = []
+    indicator_found = False
+    dbot_score = adjusted_score = 0
+
+    for threat in threats:
+        severity_level = 0
+        for block in threat.get('blockSet'):
+            if block.get('impact'):
+                threat_score: int = SEVERITY_SCORE.get(block.get('impact'), 0)
+                adjusted_score = 3 if threshold_score <= threat_score else threat_score
```

Comment:
is this taken from the original integration?

Reply:
yes